templates/toil-infra-essentials.yaml

```diff
@@ -0,0 +1,63 @@
+AWSTemplateFormatVersion: 2010-09-09
+Resources:
+  AWSKmsInfraKey:
+    Type: "AWS::KMS::Key"
+    Properties:
+      Description: !Join
+        - '-'
+        - - !Ref AWS::StackName
+          - "InfraKey"
+      KeyPolicy:
+        Version: "2012-10-17"
+        Statement:
+          -
+            Sid: "Allow administration of the key"
+            Effect: "Allow"
+            Principal:
+              AWS:
+                - !Join
+                  - ''
+                  - - 'arn:aws:iam::'
+                    - !Ref AWS::AccountId
+                    - ':root'
+                - !ImportValue us-east-1-bootstrap-TravisUserArn
+                - !ImportValue us-east-1-bootstrap-CfServiceRoleArn
+                - !ImportValue us-east-1-rna-seq-reprocessing-instance-role-v001-ToilClusterRole
+            Action:
+              - "kms:Create*"
+              - "kms:Describe*"
+              - "kms:Enable*"
+              - "kms:List*"
+              - "kms:Put*"
+              - "kms:Update*"
+              - "kms:Revoke*"
+              - "kms:Disable*"
+              - "kms:Get*"
+              - "kms:Delete*"
+              - "kms:ScheduleKeyDeletion"
+              - "kms:CancelKeyDeletion"
+            Resource: "*"
+          -
+            Sid: "Allow use of the key"
+            Effect: "Allow"
+            Principal:
+              AWS:
+                - !ImportValue us-east-1-bootstrap-TravisUserArn
+                - !ImportValue us-east-1-bootstrap-CfServiceRoleArn
+                - !ImportValue us-east-1-rna-seq-reprocessing-instance-role-v001-ToilClusterRole
```

Comment:
gives permission to use the key so you can remove the same line in the section above.

Reply:
Also this depends on the the output in `templates/toil-cluster-role-and-policy.yaml`.  Sceptre does not guarantee running templates in a certain order.  You'll need to split this PR into two PRs and manually synchronize merging them in the correct order.

Reply:
Sorry, i take that back.  i forgot that you can synchroize the deploy order with [sceptre dependencies](https://sceptre.cloudreach.com/latest/docs/stack_config.html#dependencies).  An example of that definition is here: https://github.com/Sage-Bionetworks/sandbox-infra/blob/master/config/prod/accounts.yaml#L3

Reply:
@zaro0508 in your comment above, "gives permission to use the key..." is the subject missing? the instance role does need to be able to use the key

Reply:
i just mean that setting principal here will give permission to use the kms key which is you need.